src/container/cli.c

```diff
@@ -2468,23 +2493,27 @@ dc_cont_del_attr(tse_task_t *task)
 	in->cadi_count = args->n;
 	rc = attr_bulk_create(args->n, (char **)args->names, NULL, NULL,
 			      daos_task2ctx(task), CRT_BULK_RO, &in->cadi_bulk);
-	if (rc != 0)
-		D_GOTO(cleanup, rc);
+	if (rc != 0) {
+		cont_req_cleanup(CLEANUP_RPC, &cb_args);
+		D_GOTO(out, rc);
+	}
 
 	cb_args.cra_bulk = in->cadi_bulk;
 	rc = tse_task_register_comp_cb(task, cont_req_complete,
 				       &cb_args, sizeof(cb_args));
-	if (rc != 0)
-		D_GOTO(cleanup, rc);
+	if (rc != 0) {
+		cont_req_cleanup(CLEANUP_BULK, &cb_args);
+		D_GOTO(out, rc);
+	}
 
 	crt_req_addref(cb_args.cra_rpc);
 	rc = daos_rpc_send(cb_args.cra_rpc, task);
-	if (rc != 0)
-		D_GOTO(cleanup, rc);
+	if (rc != 0) {
+		cont_req_cleanup(CLEANUP_ALL, &cb_args);
```

Comment:
daos_rpc_send() can handle the fail case's crt_req_decref(rpc), so this change looks unnecessary.

Reply:
ok